### DISCUSSION TITLE
transactional-update being aborted

### DISCUSSION THREAD

**Discussion by gefloh** [mysticaltech/terraform-hcloud-kube-hetzner#840] [Q&A] [answered]
Hello,

recently I noticed that the transactional-update is not finishing its business anymore but aborting every night on all nodes with:

```bash
Problem: the installed python310-responses-0.23.1-2.1.noarch requires 'python310-urllib3 < 2', but this requirement cannot be provided
  deleted providers: python310-urllib3-1.26.15-2.1.noarch
not installable providers: python310-urllib3_1-1.26.16-1.1.noarch[repo-oss]
 Solution 1: Following actions will be done:
  deinstallation of python310-responses-0.23.1-2.1.noarch
  deinstallation of cloud-init-23.1-1.3.aarch64
  deinstallation of cloud-init-config-MicroOS-1.1-2.10.noarch
  deinstallation of patterns-microos-cloud-5.0-70.1.aarch64
 Solution 2: deinstallation of python310-urllib3-1.26.15-2.1.noarch
 Solution 3: keep obsolete python310-urllib3-1.26.15-2.1.noarch
 Solution 4: break python310-responses-0.23.1-2.1.noarch by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/3/4/c/d/?] (c): c
2023-06-13 01:27:44 Application returned with exit status 4.
```

Obviously I have to solve this manually. The question is just: How? I am not sure what to choose here. What to do in cases like this?

Thanks!

**Comment by mysticaltech** [comment, marked as answer]
@gefloh IMHO this problem will resolve on its own when `python310-responses` will get automatically upgraded. Just wait IMHO. 

But if you want choose one of the solution proposed, just use `transactional-update pkg`. More info here https://kubic.opensuse.org/documentation/man-pages/transactional-update.8.html

**Comment by gefloh** [comment]
The problem solved itself in a different way: The cluster went completely wild today, so I deleted everything and redeployed. :-D Sometimes strange things go on, also at Hetzner with servers e.g. which are not fully deployed but stay in kind of a broken state. Anyway, thanks!

**Reply by mysticaltech** [reply]
Ok, thanks for the info. Next time please don't hesitate to share more info if things do go haywire again.

**Reply by mysticaltech** [reply]
Almost if it's for production, best to deploy in multi-location HA, so that local Hetzner problems in one location do not bring the whole thing down.

**Reply by gefloh** [reply]
HA I have, before also in different locations, yes. But now I switched to ARM completely because the prices are much better (especially considering RAM) and currently they offer ARM still only in fsn1, but I hope they will change that soon.

**Reply by mysticaltech** [reply]
Hopefully soon, indeed!
